bytes_ostream.hh

```diff
@@ -37,11 +37,11 @@ private:
     //       and chunk::frag_size refers to the currently occupied space in the chunk.
     //       After building, the first chunk::size is the whole object size, and chunk::frag_size
     //       doesn't change. This fits with managed_bytes interpretation.
```

Comment:
> of size >= 7 bytes (smaller ones fit in the inline storage of

16? max_inline_size = 15.

Reply:
And that's assuming you don't have TTL, because if you do, then you get another 12 bytes.

Reply:
Ah, such a pity. a cell block thing could deduplicate the timestamp/TTL if they are all equal (and delta compress if not).

Reply:
If we are mentioning cell blocks: remember that we don't have to look for one data structure which handles both small and big rows well. We just keep using the radix tree for sufficiently big rows (e.g. more than 128 cells or 10 kiB), but invent a single-buffer layout for smaller rows (which probably account for 99% of the cases where the optimization is actually significant).

Reply:
> Ah, such a pity. a cell block thing could deduplicate the timestamp/TTL if they are all equal (and delta compress if not).

@avikivity - https://github.com/scylladb/scylladb/issues/2954 ?

Reply:
> > Ah, such a pity. a cell block thing could deduplicate the timestamp/TTL if they are all equal (and delta compress if not).
> 
> @avikivity - #2954 ?

Yes, that's the same idea. It's an obvious optimization.

Reply:
I wanted a radix tree of cell blocks.

Reply:
> I wanted a radix tree of cell blocks.

I know, that's why I'm bringing up other ideas before we get too attached to this one.